_posts/2024-12-05-release-notes-3.6.2.md

```diff
@@ -0,0 +1,328 @@
+
+```scala
+//> using options -experimental
+trait AppConfig:
+  def get(key: String): Option[String]
+val config: AppConfig = ???
+
+val Some(appVersion) = config.get("appVersion").runtimeChecked
+```
+
+# Other notable changes
+
+## Switch mapping of context bounds to using clauses
+
+Until Scala 3.6 context bound parameters were always desugared to `implicit` arguments, starting with Scala 3.6 these would be mapped to `using` parameters instead.
+This change should not affect the majority of users, however, it can lead to differences in how implicits are resolved.
+Resolution of implicits can slightly differ depending on whether we're requesting them using `implicit` or `using` parameter, or depending on whether they were defined using `implicit` or `given` keywords. The special behaviours were introduced to smoothen migration from Scala 2 to brand new implicits resolution in Scala 3.
+This change might also affect some of the projects that use compiler plugins or macros to inspect the implicit argument lists of the function calls - these might require some minor fixes, eg. when filtering symbols by their flags.
+
+## Work on a better scheme for given prioritization
+
+In the [Scala 3.5.0 release notes](https://scala-lang.org/blog/2024/08/22/scala-3.5.0-released.html) we've announced upcoming changes to givens, due to their peculiar problem with prioritization. Currently, the compiler always tries to select the instance with the most specific subtype of the requested type. In the future, it would change to always selecting the instance with the most general subtype that satisfies the context-bound.
+
+Starting from Scala 3.6, code whose behaviour can differ between new and old rules (ambiguity on new, passing on old, or vice versa) will emit warnings, but the old rules will still be applied.
+Running the compiler with `-source:3.5` will allow you to temporarily keep using the old rules; with `-source:3.7` or `-source:future` the new scheme will be used.
+
+For the detailed motivation of changes with examples of code that will be easier to write and understand, see our recent blog post - [Upcoming Changes to Givens in Scala 3.7]({{ site.baseurl }}/2024/08/19/given-priority-change-3.7.html).
+
+## Require named arguments for Java-defined annotations
+
+Java-defined annotations don't have an exact constructor representation. The compiler previously relied on the order of the fields to create annotation instance. One possible issue with this representation is the reordering of the fields.
+Let's take the following example:
+
+```scala
+  public @interface Annotation {
+    int a() default 41;
+    int b() default 42;
+  }
+```
+
+Reordering the fields is binary-compatible but it might affect the meaning of `@Annotation(1)`
+Starting from Scala 3.6, named arguments are required for Java-defined annotations that define multiple parameters. Java defined annotations with a single parameter named `value` can still be used anonymously.
```

Comment:
This wording seems to suggest that annotations are supposed to be used anonymously rather than their arguments, which is slightly confusing. Also I think `unnamed` would be clearer than `anonymous` in this context.
And actually unnamed arguments might also work for annotations with multiple parameters given that only one argument is passed to the annotation's 'pseudo-constructor' (the one corresponding to `value` parameter) and the other parameters have a default value. e.g.

```java
import java.lang.annotation.*;

public @interface MyAnnot {
    String value();
    String value2() default "default";
}
```

```scala
@MyAnnot("abc")
class Foo
```
Just please pick some better names if you're going to reuse this example 😉

Reply:
Don't you this example might be confusing to reader? In this case depening on number of used parameters in the example you might skip names if only 1 param is used, or all required to name all of them to otherwise.

Reply:
Actually it might be usefull to explain the rules better, I've updated the example